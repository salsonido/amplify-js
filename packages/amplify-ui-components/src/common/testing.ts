import {
	boolean as booleanKnob,
	select as selectKnob,
	text as textKnob,
} from '@storybook/addon-knobs';

export const pixelThreshold = 10;

export const selectors = {
	// amplify-form-field
	formFieldDescription: '[data-test=form-field-description]',
};

export const knobs = {
	labelKnob(labelPlaceholder: string) {
		return textKnob('Label', labelPlaceholder);
	},

	descriptionKnob(descriptionPlaceholder: string) {
		return textKnob('Description', descriptionPlaceholder);
	},

	hintKnob(hintPlaceholder: string) {
		return textKnob('Hint', hintPlaceholder);
	},

	inputTypeKnob(inputTypePlaceholder: string) {
		return selectKnob(
			'Type',
			[
				'date',
				'email',
				'number',
				'password',
				'search',
				'tel',
				'text',
				'url',
				'time',
			],
			inputTypePlaceholder
		);
	},

	placeholderKnob(placeholderPlaceholder: string) {
		return textKnob('Placeholder', placeholderPlaceholder);
	},

	buttonTypeKnob(buttonTypePlaceholder: string) {
		return selectKnob(
			'Button type',
			['button', 'submit', 'reset'],
			buttonTypePlaceholder
		);
	},

	buttonTextKnob(buttonTextPlaceholder: string) {
		return textKnob('Button text', buttonTextPlaceholder);
	},

	toggleKnob(toggleDefault: boolean) {
		return booleanKnob('Toggle', toggleDefault);
	},

	disabledKnob(disabledDefault: boolean) {
		return booleanKnob('Disabled', disabledDefault);
	},

	formSectionHeaderKnob(headerPlaceholder: string) {
		return textKnob('Header text', headerPlaceholder);
	},

	formSectionSubmitKnob(submitPlaceholder: string) {
		return textKnob('Submit button text', submitPlaceholder);
	},

	tooltipAutoShowKnob(autoShowDefault: boolean) {
		return booleanKnob('Tooltip autoshow', autoShowDefault);
	},
};
