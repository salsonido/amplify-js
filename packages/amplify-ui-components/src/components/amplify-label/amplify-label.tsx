import { Component, Prop, h } from '@stencil/core';

/**
 * @slot (default) - Label content
 */
@Component({
	tag: 'amplify-label',
	styleUrl: 'amplify-label.scss',
})
export class AmplifyLabel {
	/** Reflects the value of the for content property of html element */
	@Prop() htmlFor: string;

	render() {
		return (
			<label class="label" htmlFor={this.htmlFor}>
				<slot />
			</label>
		);
	}
}
