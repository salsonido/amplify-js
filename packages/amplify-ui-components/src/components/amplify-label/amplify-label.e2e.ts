import { E2EPage, newE2EPage } from '@stencil/core/testing';
import { pixelThreshold } from '../../common/testing';

describe('amplify-label e2e:', () => {
	let page: E2EPage;

	beforeEach(async () => {
		page = await newE2EPage();
		await page.setContent('<amplify-label>This is a label</amplify-label>');
	});

	it('renders with text `This is a label`', async () => {
		const element = await page.find('amplify-label');
		expect(element).not.toBeNull();

		const screenshot = await page.compareScreenshot('amplify-label', {
			fullPage: true,
		});
		expect(screenshot).toMatchScreenshot({
			allowableMismatchedPixels: pixelThreshold,
		});
	});
});
