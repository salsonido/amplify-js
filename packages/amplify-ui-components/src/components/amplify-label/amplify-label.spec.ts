import { newSpecPage } from '@stencil/core/testing';
import { AmplifyLabel } from './amplify-label';

describe('amplify-label spec:', () => {
	describe('Render logic ->', () => {
		it('should render a label by default', async () => {
			const page = await newSpecPage({
				components: [AmplifyLabel],
				html: `<amplify-label></amplify-label>`,
			});

			expect(page.root).toMatchSnapshot();
		});

		it('should render only class `amplify-ui--label`', async () => {
			const page = await newSpecPage({
				components: [AmplifyLabel],
				html: `<amplify-label></amplify-label>`,
			});

			expect(page.root).toMatchSnapshot();
		});
	});
});
