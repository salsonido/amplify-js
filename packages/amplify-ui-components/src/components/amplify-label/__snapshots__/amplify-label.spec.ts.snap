// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-label spec: Render logic -> should render a label by default 1`] = `
<amplify-label>
  <!---->
  <label class="label"></label>
</amplify-label>
`;

exports[`amplify-label spec: Render logic -> should render only class \`amplify-ui--label\` 1`] = `
<amplify-label>
  <!---->
  <label class="label"></label>
</amplify-label>
`;
